src/OpenTelemetry/Trace/ActivityProcessor.cs

```diff
@@ -80,14 +82,24 @@ public void Dispose()
 
         protected virtual void Dispose(bool disposing)
         {
-            try
+            if (this.disposed)
             {
-                this.ShutdownAsync(CancellationToken.None).GetAwaiter().GetResult();
+                return;
             }
-            catch (Exception ex)
+
+            if (disposing)
             {
-                OpenTelemetrySdkEventSource.Log.SpanProcessorException(nameof(this.Dispose), ex);
+                try
+                {
+                    this.ShutdownAsync(CancellationToken.None).GetAwaiter().GetResult();
+                }
+                catch (Exception ex)
+                {
+                    OpenTelemetrySdkEventSource.Log.SpanProcessorException(nameof(this.Dispose), ex);
+                }
             }
+
+            this.disposed = true;
```

Comment:
Do we need set `disposed` to `true` if exception happened in the call to `ShutdownAsync`, or just throw the exception out?

Reply:
I think we need to set to true as dispose is not supposed to fail / retry.